src/components/Task/Task.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  type Dispatch,
+  type FC,
+  type SetStateAction,
+  useCallback,
+  useEffect,
+  useRef,
+} from 'react'
+
+import { updateTaskThunk } from '../../redux/slices/toDoSlice'
+import { useDispatch, useSelector } from '../../redux/store'
+import type { Task as TaskType } from '../../types'
+import { Loader } from '../Loader'
+import { Cancel } from './Cancel'
+import { Check } from './Check'
+import { EditTask } from './EditTask'
+import { Pencil } from './Pencil'
+import styles from './Task.module.css'
+import { TaskCheckbox } from './TaskCheckbox'
+import { Trash } from './Trash'
+
+interface Props {
+  task: TaskType;
+  editId: number | null;
+  setEditId: Dispatch<SetStateAction<number | null>>;
+}
+
+export const Task: FC<Props> = ({ task, editId, setEditId }) => {
+  const editInputRef = useRef(null as HTMLInputElement | null)
+  const isEdit = editId === task.id
+  const dispatch = useDispatch()
+  const isLoading = useSelector((state) =>
+    state.tasks.todoLoadingIds.includes(task.id),
+  )
+
+  const submit = useCallback(() => {
+    dispatch(
+      updateTaskThunk({
+        taskId: task.id,
+        taskValue: editInputRef.current?.value,
+      }),
+    ).then(() => setEditId(null))
+  }, [dispatch, task.id, setEditId])
+
+  useEffect(() => {
+    const listener = (event: KeyboardEvent) => {
+      if (isEdit) {
+        if (event.key === 'Enter') {
+          if (editInputRef.current?.value !== task.todo) {
+            submit()
+          } else {
+            setEditId(null)
+          }
```

Comment:
```suggestion
          submit()
```

Reply:
Could I write here like this?  Just optionaly for information

```suggestion

onClick={submit}

instead of this:

onClick={()=> {
   submit()
}}

```